go/libraries/doltcore/sqle/dtables/branches_table.go

```diff
@@ -34,34 +35,51 @@ var _ sql.ReplaceableTable = (*BranchesTable)(nil)
 // BranchesTable is a sql.Table implementation that implements a system table which shows the dolt branches
 type BranchesTable struct {
 	ddb *doltdb.DoltDB
+	all bool
 }
 
 // NewBranchesTable creates a BranchesTable
 func NewBranchesTable(_ *sql.Context, ddb *doltdb.DoltDB) sql.Table {
-	return &BranchesTable{ddb}
+	return &BranchesTable{ddb, false}
+}
+
+// NewAllBranchesTable creates a BranchesTable with remote refs included
+func NewAllBranchesTable(_ *sql.Context, ddb *doltdb.DoltDB) sql.Table {
+	return &BranchesTable{ddb, true}
 }
 
 // Name is a sql.Table interface function which returns the name of the table which is defined by the constant
 // BranchesTableName
 func (bt *BranchesTable) Name() string {
+	if bt.all {
+		return doltdb.AllBranchesTableName
+	}
 	return doltdb.BranchesTableName
 }
 
 // String is a sql.Table interface function which returns the name of the table which is defined by the constant
 // BranchesTableName
 func (bt *BranchesTable) String() string {
+	if bt.all {
+		return doltdb.AllBranchesTableName
+	}
 	return doltdb.BranchesTableName
 }
 
 // Schema is a sql.Table interface function that gets the sql.Schema of the branches system table
 func (bt *BranchesTable) Schema() sql.Schema {
```

Comment:
Seems like it would be nice to have a column that indicated if a branch was a local branch or remote branch so you could quickly filter to just remote branches without having to do string prefix matching on the name.